src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -434,7 +467,16 @@ internal static async IAsyncEnumerable<DocumentId> GetChangedDocumentsAsync(Proj
                 yield return documentId;
             }
 
-            if (!HasChangesThatMayAffectSourceGenerators(oldProject.State, newProject.State))
+            // Given the following assumptions:
+            // - source generators are deterministic,
+            // - metadata references and compilation options have not changed (TODO -- need to check),
+            // - source documents have not changed,
+            // - additional documents have not changed,
+            // - analyzer config documents have not changed,
+            // the outputs of source generators will not change.
+
+            if (!newProject.State.AdditionalDocumentStates.HasAnyStateChanges(oldProject.State.AdditionalDocumentStates) &&
+                !newProject.State.AnalyzerConfigDocumentStates.HasAnyStateChanges(oldProject.State.AnalyzerConfigDocumentStates))
```

Comment:
Doesn't this still need to check document states? The above states the assumption "source documents have not changed", but thats not checked. In fact, the loop above could have yielded them.

Reply:
Good point. I think you're right.

Reply:
Fixed and added testing of source generated files.